Add Quick Start Guide to NVDA User Guide

Qchristensen opened the pull request:
<!-- Please read and fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/blob/master/devDocs/githubPullRequestTemplateExplanationAndExamples.md
Please also note that the NVDA project has a Citizen and Contributor Code of Conduct which can be found at https://github.com/nvaccess/nvda/blob/master/CODE_OF_CONDUCT.md. NV Access expects that all contributors and other community members read and abide by the rules set out in this document while participating or contributing to this project. This includes creating or commenting on issues and pull requests. 

Please initially open PRs as a draft. See https://github.com/nvaccess/nvda/wiki/Contributing
-->

### Link to issue number:
NA
### Summary of the issue:

User requests for a more in-depth guide to help new users get started with NVDA.  The guide has been written with the intention of including it in the User Guide, but also being able to easily separate it out into a standalone document where needed.

The quick start guide also helps bring users onboard in such a way as to give them enough information to get started, and also include information about other resources (eg training material and user groups), to ensure they are able to find additional support if needed.
 
### Description of user facing changes

- Move System Requirements to become section 1.2
- Make new "Quick Start Guide" section 2, with sections on downloading NVDA, installing NVDA and Running NVDA (and smaller topics in the quick start guide on preferences, getting help, community.
- The idea is that the Quick Start Guide will also be able to be used outside the user guide as a standalone document.
- Section 3 "More setup options" includes most of the original section 3, but renamed as "Getting and setting up NVDA" could be confusing next to "Quick start guide".  Most of the text is unchanged, as although slightly repetitious after the quick start guide, the quick start guide walks deeply through (default) options where "More setup options" covers other details, such as information on portable copies.


I appreciate this is a large documentation update and I apologise in advance to the translators.  I am deeply grateful for all your work!

### Description of development approach

Early work on this was included in the Switching from Jaws to NVDA Wiki some time ago and has been favourably commented on.  The text has been drafted several times with the intention of complementing the remainder of the User Guide.

### Testing strategy:

### Known issues with pull request:

Nonw

### Change log entries:
New features
Added Quick Start Guide to User Guide

### Code Review Checklist:

<!--
This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review of this pull-request.
Check items to confirm you have thought about the relevance of the item.
Where items are missing (eg unit / system tests), please explain in the PR.
To check an item `- [ ]` becomes `- [x]`, note spacing.
You can also check the checkboxes after the PR is created.
A detailed explanation of this checklist is available here:
https://github.com/nvaccess/nvda/blob/master/devDocs/githubPullRequestTemplateExplanationAndExamples.md#code-review-checklist
-->

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

Qchristensen committed:
Add Quick Start Guide to NVDA User Guide

- Move System Requirements to become section 1.2
- Make new "Quick Start Guide" section 2, with sections on downloading NVDA, installing NVDA and Running NVDA (and smaller topics in the quick start guide on preferences, getting help, community.
- The idea is that the Quick Start Guide will also be able to be used outside the user guide as a standalone document.
- Section 3 "More setup options" includes most of the original section 3, but renamed as "Getting and setting up NVDA" could be confusing next to "Quick start guide".  Most of the text is unchanged, as although slightly repetitious after the quick start guide, the quick start guide walks deeply through (default) options where "More setup options" covers other details, such as information on portable copies.

lukaszgo1 commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
+If you plan to always use NVDA on this computer, you will want to choose to install NVDA.
+Installing NVDA will allow for additional functionality such as automatic starting after sign-in, the ability to read the Windows sign-in and [secure screens #SecureScreens].
+These cannot be done with portable and temporary copies.
+For full details of the limitations when when running a portable or temporary copy of NVDA, please see [Portable and temporary copy restrictions PortableAndTemporaryCopyRestrictions].
```
Double 'when"
```suggestion
For full details of the limitations when running a portable or temporary copy of NVDA, please see [Portable and temporary copy restrictions PortableAndTemporaryCopyRestrictions].
```

lukaszgo1 commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
+|| Name | Desktop key | Laptop key | Touch | Description |
+| Window title | NVDA+t | NVDA+t | | Reports the title of the currently active window. Pressing twice will spell the information. Pressing three times will copy it to the clipboard |
+| Report focus | NVDA+tab | NVDA+tab | | Reports the title of the currently active window. Pressing twice will spell the information. Pressing three times will copy it to the clipboard |
+| Read window | NVDA+b | NVDA+b | | Reports the title of the currently active window. Pressing twice will spell the information. Pressing three times will copy it to the clipboard |
```
The description here is incorrect.

Qchristensen replied:
So it was, for some reason it didn't copy the text when I selected the other descriptions so pasted the same thing again.  I've updated it now.

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Łukasz Golonka <lukasz.golonka@mailbox.org>

Qchristensen committed:
Update userGuide.t2t

Fixed description for items in the "Reporting location and other information"

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -34,6 +34,13 @@ Major highlights include:
 -
 
++ System Requirements +[SystemRequirements]
+- Operating Systems: all 32-bit and 64-bit editions of Windows 7, Windows 8, Windows 8.1, Windows 10, Windows 11, and all Server Operating Systems starting from Windows Server 2008 R2.
+ - For Windows 7, NVDA requires Service Pack 1 or higher.
+ - For Windows Server 2008 R2, NVDA requires Service Pack 1 or higher.
+- at least 150 MB of storage space.
+-
```
Looks like the list formatting here is incorrect - good chance to fix it up.

```suggestion
- Operating Systems: all 32-bit and 64-bit editions of Windows 7, Windows 8, Windows 8.1, Windows 10, Windows 11, and all Server Operating Systems starting from Windows Server 2008 R2.
  - For Windows 7, NVDA requires Service Pack 1 or higher.
  - For Windows Server 2008 R2, NVDA requires Service Pack 1 or higher.
  -
- at least 150 MB of storage space.
-

```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
+
+This quick start guide contains three main sections: downloading, initial setup, and running NVDA.
+These are followed by information on adjusting preferences, participating in the community and getting help.
+The information in this guide is condensed from other parts of the [NVDA User Guide ].
```
Is this meant to be a link?

```suggestion
The information in this guide is condensed from other parts of the NVDA User Guide.
```

Qchristensen replied:
I was going to make it a link, especially for when we copy it out of the user guide, but then I wasn't sure how best to do it - perhaps to https://www.nvaccess.org/files/nvda/documentation/userGuide.html and then I was thinking about the other links in this section to other parts of the user guide?

seanbudd replied:
I think it's fine to not link it - it is strange for there to be a self-referential link.

This might be better:

```suggestion
The information in this guide is condensed from other parts of this document.
```

Qchristensen replied:
Should we say "this document", or "this NVDA User Guide" - that way when we do extract it as its own document, someone reading it can at least pick up that we meant the NVDA User Guide not this quick start guide (when we inevitably forget to link it there).

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
++	Activate the "download" button
++ The browser may or may not prompt for an action after downloading, and then start the download
++ Depending on the browser, the file may run automatically after it downloads
++ If the file needs to be manually launched, press `alt+n` to move to the notification area, then `alt+r` to run the file (or the steps for your browser)
```
Monospace formatting must use double backticks ``

```suggestion
+ If the file needs to be manually launched, press ``alt+n`` to move to the notification area, then ``alt+r`` to run the file (or the steps for your browser)
```

seanbudd replied:
This is still an issue across the entire document

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
++++ Reading text:
+|| Name | Desktop key | Laptop key | Touch | Description |
+| Say all | NVDA+down arrow | NVDA+a | | Starts reading from the current position, moving it along as it goes |
+| Read current line | NVDA+up arrow | NVDA+l | | Reads the line. Pressing twice spells the line. Pressing three times spells the line using character descriptions (Alpha, Bravo, Charlie, etc) |
```
Key commands should be lowerCamelCase and encapsulated as monospace codeblocks

```suggestion
| Read current line | ``NVDA+upArrow`` | ``NVDA+l`` | | Reads the line. Pressing twice spells the line. Pressing three times spells the line using character descriptions (Alpha, Bravo, Charlie, etc) |
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +72,223 @@ This applies to both original and modified copies of this software, plus any der
+We recommend starting with the "Basic Training for NVDA module".
+This module covers concepts from getting started up to browsing the web and using object navigation.
+It is available in:
+-	[Electronic text https://www.nvaccess.org/product/basic-training-for-nvda-ebook/], which includes Word DOCX, Web page HTML, eBook ePub and Kindle KFX formats.
+-	[Human-read, MP3 audio https://www.nvaccess.org/product/basic-training-for-nvda-downloadable-audio/]
+-	[Hard-copy UEB Braille https://www.nvaccess.org/product/basic-training-for-nvda-braille-hard-copy/] with delivery included anywhere in the world.
```
Lists must be terminated with a final `-`

```suggestion
- [Electronic text https://www.nvaccess.org/product/basic-training-for-nvda-ebook/], which includes Word DOCX, Web page HTML, eBook ePub and Kindle KFX formats.
- [Human-read, MP3 audio https://www.nvaccess.org/product/basic-training-for-nvda-downloadable-audio/]
- [Hard-copy UEB Braille https://www.nvaccess.org/product/basic-training-for-nvda-braille-hard-copy/] with delivery included anywhere in the world.
-
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -142,6 +326,25 @@ Pressing Continue will create the portable copy.
+Using the temporary copy of NVDA is also an option (e.g. for demonstration purposes), though starting NVDA in this way each time can become very time consuming.
+
+Apart from the inability to automatically start during and/or after log-on, the portable and temporary copies of NVDA also have the following restrictions:
+-	The inability to interact with applications running with administrative privileges, unless of course NVDA itself has been run also with these privileges (not recommended).
+-	The inability to read User Account Control (UAC) screens when trying to start an application with administrative privileges.
+-	Windows 8 and later: the inability to support input from a touchscreen.
+-	Windows 8 and later: the inability to provide features such as browse mode and speaking of typed characters in Windows Store apps.
```
I'm not sure why these tabs are included - but it is inconsistent with other lists

```suggestion
- The inability to interact with applications running with administrative privileges, unless of course NVDA itself has been run also with these privileges (not recommended).
- The inability to read User Account Control (UAC) screens when trying to start an application with administrative privileges.
- Windows 8 and later: the inability to support input from a touchscreen.
- Windows 8 and later: the inability to provide features such as browse mode and speaking of typed characters in Windows Store apps.
```

seanbudd replied:
This is still an issue across the entire document

Qchristensen replied:
I've replaced all the single ` with double ``.  Sorry, still getting used to this formatting!

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Thanks Sean - I didn't check that part, just moved it.

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Thanks!

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen commented:
Thanks @seanbudd for picking up those formatting issues!

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update userGuide.t2t

Changed single ` for double ``

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -34,6 +34,15 @@ Major highlights include:
 - The ability to highlight the system focus
 -
 
++ System Requirements +[SystemRequirements]
```
This used to be a top level header, which means every subsequent item below this used to be under "1. Introduction" now are under "2. System Requirements". I suggest system requirements become a subsection under introduction

```suggestion
++ System Requirements ++[SystemRequirements]
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,224 @@ This applies to both original and modified copies of this software, plus any der
+The information in this guide is condensed from other parts of the NVDA User Guide.
+Please refer to the full User Guide for more detailed information on each topic.
+
+++ Downloading NVDA +[GettingAndSettingUpNVDA]
```
The number of `+`s must be the same on either side, this is causing several headings to not render properly.

```suggestion
++ Downloading NVDA ++[GettingAndSettingUpNVDA]
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,224 @@ This applies to both original and modified copies of this software, plus any der
+
+The setup file is named "nvda_2022.1.exe" or similar.
+The year and version changes between updates to reflect the current release.
++	Run the downloaded file.
```
This list is also broken

```suggestion
- Run the downloaded file.
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,224 @@ This applies to both original and modified copies of this software, plus any der
+These steps walk through the most common setup options.
+For more details on the options available, please see [Installation options InstallingNVDA].
+
++	From the launcher, ensure the checkbox to agree to the license is checked.
```
Same with this list

seanbudd commented:
I would suggest checking the build output.

You can do this by:
1. Click "Show all checks" (ctrl+f to ifnd) on this page 
1. Click details
1. Navigate to artefacts (e.g. for the latest build: https://ci.appveyor.com/project/NVAccess/nvda/builds/44292974/artifacts)
1. download `output/userGuide.html`
1. Open `output/userGuide.html` in your browser.

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,224 @@ This applies to both original and modified copies of this software, plus any der
+Music plays while a temporary copy of NVDA loads.
+Once loaded, NVDA will speak throughout the rest of the process.
++	The NVDA Launcher window appears with the license agreement.
+Press ``down arrow`` to read the license agreement if desired.
```
```suggestion
Press ``downArrow`` to read the license agreement if desired.
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,224 @@ This applies to both original and modified copies of this software, plus any der
+If you plan to always use NVDA on this computer, you will want to choose to install NVDA.
+Installing NVDA will allow for additional functionality such as automatic starting after sign-in, the ability to read the Windows sign-in and [secure screens #SecureScreens].
+These cannot be done with portable and temporary copies.
+For full details of the limitations when running a portable or temporary copy of NVDA, please see [Portable and temporary copy restrictions PortableAndTemporaryCopyRestrictions].
```
```suggestion
For full details of the limitations when running a portable or temporary copy of NVDA, please see [Portable and temporary copy restrictions #PortableAndTemporaryCopyRestrictions].
```

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Fix formatting of Quick Start Guide

Hopefully fixed all the section headings and keystroke highlighting.  Also renamed the original "Getting Started with NVDA" section to "Using NVDA" since the name was too close to the quick start guide section name.  Note the quick start guide does have a topic called "Running NVDA" but I think they are different enough and one is a chapter heading and one a topic in the quick start guide.  Happy for feedback though!

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update formatting of user guide

Removed several tabs in lists, and changed keystrokes to single word, camel case (eg "down arrow" to "downArrow")

Qchristensen committed:
Update user_docs/en/userGuide.t2t

Co-authored-by: Sean Budd <sean@nvaccess.org>

Qchristensen committed:
Update userGuide.t2t

Fixed several links missing #

Qchristensen committed:
Update userGuide.t2t

Fixed a few more things I noticed after looking at the output.  Specifically several URLs, and the steps in 2.2.1 and 2.2.2 which both had sub-lists in the middle, as well as the topic names in "Running NVDA".

Qchristensen committed:
Update userGuide.t2t

Fix table headings I had a semi-colon on which broke the heading.

Qchristensen committed:
Update userGuide.t2t

Fixed step numbering in 2.2.2 Steps for installing NVDA.
Fixed heading for 2.3.8 web navigation

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,226 @@ This applies to both original and modified copies of this software, plus any der
+The full NVDA user guide has all the NVDA commands, split up into different sections for reference.
+The tables of commands are also available in the "Commands Quick Reference".
+The "Basic Training for NVDA" NVDA training module has each command in more depth with step-by-step activities.
+"Basic Training for NVDA" is available from the [NV Access Shop http://www.nvaccess.org/shop]
```
Add missing full stop

```suggestion
"Basic Training for NVDA" is available from the [NV Access Shop http://www.nvaccess.org/shop].
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,226 @@ This applies to both original and modified copies of this software, plus any der
++++ Starting and stopping NVDA +++[StartingAndStoppingNVDA]
+|| Name | Desktop key | Laptop key | Description |
+| Start NVDA | ``control+alt+n`` | ``control+alt+n`` | Starts or restarts NVDA |
+| Exit NVDA | ``NVDA+q``, then ``enter`` | ``NVDA+q``, then enter | Exits NVDA |
```
```suggestion
| Exit NVDA | ``NVDA+q``, then ``enter`` | ``NVDA+q``, then ``enter`` | Exits NVDA |
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -65,46 +74,226 @@ This applies to both original and modified copies of this software, plus any der
+
+If installing NVDA directly from the downloaded NVDA launcher, press the Install NVDA button.
+If you have already closed this dialog or are wanting to install from a portable copy, please choose the Install NVDA menu item found under Tools in the NVDA menu.
+The installation dialog that appears will confirm whether you wish to install NVDA and will also tell you whether this installation will be updating a previous install.
```
This paragraph is quite large.

```suggestion

The installation dialog that appears will confirm whether you wish to install NVDA and will also tell you whether this installation will be updating a previous install.
```

seanbudd commented on user_docs/en/userGuide.t2t:
```diff
@@ -123,15 +312,15 @@ This option is enabled by default for fresh installations.
 
 ++ Creating a Portable Copy ++[CreatingAPortableCopy]
-If creating a portable copy directly from the NVDA download  package, simply press the Create Portable Copy button.
+If creating a portable copy directly from the NVDA download  package, press the Create Portable Copy button.
```
might as well fix this extra space too

```suggestion
If creating a portable copy directly from the NVDA download package, press the Create Portable Copy button.
```

seanbudd reviewed (commented):
Excellent work @Qchristensen , excited to get this in. I've read through this thoroughly now and looks good to me. Only a few minor suggestions.

feerrenrut reviewed (approved):
Content looks good. Good work!

seanbudd committed:
Apply suggestions from review

seanbudd reviewed (approved):
I've just committed these changes as they were minor.
Going to merge when the build succeeds. 🎉
